### PR TITLE
Add user that has permissions to pull backups of supersetDB

### PR DESCRIPTION
**Pull request by maulikjs**

### CODE REVIEW

File: kfdefs/overlays/prod/dh-prod-superset/supersetdb-deployment-pvc-backup.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                  psql '-c' "GRANT SELECT ON ALL TABLES IN SCHEMA public TO bareos;";
+                  psql --dbname=superset -c "GRANT SELECT ON ALL TABLES IN SCHEMA public TO bareos;";
+                  psql '-c' "GRANT USAGE ON SCHEMA public TO bareos;";
+                  psql '-c' "GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA public TO bareos;";
+        env:
+        - name: BACKUPS_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              key: backups-password
+              name: prod-supersetdb-secret
```
Comment (rimolive) [inline kfdefs/overlays/prod/dh-prod-superset/supersetdb-deployment-pvc-backup.yaml:30]: No newline

File: kfdefs/overlays/prod/dh-prod-superset/kustomization.yaml [hunk 2]
```diff
@@ -115,3 +115,6 @@ patchesJson6902:
       kind: Route
       name: superset
       version: v1
+
+patchesStrategicMerge:
+  - ./supersetdb-deployment-pvc-backup.yaml
```
Comment (rimolive) [inline kfdefs/overlays/prod/dh-prod-superset/kustomization.yaml:120]: No newline

**Review by rimolive — CHANGES_REQUESTED** [review]

**Review by anishasthana — APPROVED** [review]

File: kfdefs/overlays/prod/dh-prod-superset/supersetdb-deployment-pvc-backup.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      - name: postgresql
+        lifecycle:
+          postStart:
+            exec:
+              command:
+                - /bin/sh
+                - '-c'
+                - >
+                  sleep 5;
+                  psql '-c' "CREATE USER bareos WITH PASSWORD '$BACKUPS_PASSWORD';";
```
Comment (accorvin) [inline kfdefs/overlays/prod/dh-prod-superset/supersetdb-deployment-pvc-backup.yaml:20]: Will this fail on subsequent deployments of the database pod?

Comment (maulikjs) [reply kfdefs/overlays/prod/dh-prod-superset/supersetdb-deployment-pvc-backup.yaml:20]: Well yeah but if a user already exists it does nothing.

**Review by rimolive — APPROVED** [review]

**Pull request merged**